sqlglot/dataframe/sql/functions.py

```diff
@@ -1113,7 +1113,7 @@ def reverse(col: ColumnOrName) -> Column:
 
 
 def flatten(col: ColumnOrName) -> Column:
-    return Column.invoke_anonymous_function(col, "FLATTEN")
+    return Column.invoke_expression_over_column(None, expression.Flatten, expressions=[col])
```

Comment:
is this semantically the same for spark?? if not it’s may not be right

Reply:
I think the only difference is that now we have a dedicated `Flatten` expression as opposed to an `Anonymous` one. For reference, Spark's `FLATTEN` is documented in https://spark.apache.org/docs/latest/api/sql/index.html#flatten. We also have a couple of dataframe tests for this (see test_flatten).

Reply:
Was using [this diff](https://github.com/tobymao/sqlglot/pull/2417#discussion_r1362516004), but flatten is a list flatten function not the same as the snowflake function.

Reply:
Yeah they definitely don't have the same semantics and syntax, I think though Toby was asking whether or not this is breaking in terms of SQLGlot AST representation?

Reply:
flatten should be list flatten, let’s make another node that is snowflakes variant

Reply:
so have a exp.SnowflakeFlatten?

Reply:
call it something more generic but yes, dunno what’s a good phrase but don’t use the word snowflake, describe its func

Reply:
snowflake's flatten is basically an `explode`, right? perhaps we can reuse that node instead? cc @tobymao

reference: https://docs.snowflake.com/en/sql-reference/functions/flatten

> Flattens (explodes) compound values into multiple rows.

Reply:
yea, then just map snowflake flatten to explode, that could help with transpilation

Reply:
Mapped it to explode